src/features/defi/providers/osmosis/components/OsmosisManager/Lp/Deposit/components/Deposit.tsx

```diff
@@ -97,7 +97,7 @@ export const Deposit: React.FC<DepositProps> = ({
   const userAddress = useMemo(() => accountId && fromAccountId(accountId).account, [accountId])
   const toast = useToast()
 
-  const getDepositFeeEstimate = useCallback(async (): Promise<string | undefined> => {
+  const getDepositFeeEstimateCryptoBaseUnit = useCallback(async (): Promise<string | undefined> => {
```

Comment:
Obviously beyond the scope of this PR, but I'd prefer if we leveraged the type system for this as it's cleaner and more functional IMO. The long variable names make the code less readable, and I'm having [Apple SDK flashbacks](https://developer.apple.com/documentation/coremedia/1489454-cmmetadataformatdescriptioncreat) from the Objective-C days.


Take for example a function definition using the current naming convention:
<img width="960" alt="Screenshot 2023-02-14 at 3 43 01 PM" src="https://user-images.githubusercontent.com/62026038/218880187-57766755-4ec0-4e40-bce1-d80e28296c24.png">
That's a lot of text both in the function signature and implementation.


If we instead use a type alias, we can save quite a bit of text in the function implementation and can be sure at a glance that the vernacular is enforced via the parameter and return types.
<img width="828" alt="Screenshot 2023-02-14 at 3 43 22 PM" src="https://user-images.githubusercontent.com/62026038/218880886-5702acb8-f55d-48d9-8df7-954217b8d203.png">


Additionally, documentation to be inlined wherever the type alias is used. It isn't completely clear at first glance what `CryptoBaseUnit` actually means, so this will improve the DX for new developers considerably.
<img width="824" alt="Screenshot 2023-02-14 at 3 50 52 PM" src="https://user-images.githubusercontent.com/62026038/218881342-161e44bd-1d44-4bcf-8732-3176824b61c6.png">


Since TypeScript provides documentation on mouseover, this gives the option to use terse type aliases with no loss of information and save a lot of text. 
<img width="724" alt="Screenshot 2023-02-14 at 3 56 11 PM" src="https://user-images.githubusercontent.com/62026038/218881938-bce0b254-e0b6-4601-b79c-c2432804fea0.png">


Of the two approaches, the second is more readable and more informative IMO.
<img width="952" alt="Screenshot 2023-02-14 at 3 57 54 PM" src="https://user-images.githubusercontent.com/62026038/218882167-02038199-e121-4bfc-83dd-523078092a77.png">

Reply:
Yeah, that might be over-expressive Objective-C style, but using an alias/flavor is not going to solve much/all of our problems:

- `Additionally, documentation to be inlined wherever the type alias is used.` that's not exactly true, your IDE may help you and hint you WRT the type but

  1.  You might not be paying attention to the actual name of the type, and think the precision you've just passed is fine, since the compiler doesn't complain (I know my LSP setup takes more time to pop-up the function signature then it takes me to actually write the nth arg, and that's when it works)
  2. Consumption when reading a GitHub (or Graphite, CLI, or any tool you're using for the job really) code review won't give you the types
  3. `Since TypeScript provides documentation on mouseover` what's a mouse

Reply:
1.) I agree that it would be great to be able to enforce the precision consistency at compile time, but adding the precision to the variable name has the same issue.
2.) The type with precision info shouldn't ever be so far away from the function signature or variable declaration that it's cumbersome to find in these situations, should it?
3.) Sorry, _"la souris"_

I worked on some scientific computing applications back in the day, and this is a pretty common pattern in situations where you have to keep track of lots of variables with different units but don't want to go as far as rolling a class to handle the conversions and math operations.